activesupport/lib/active_support/concurrency/load_interlock_aware_monitor.rb

```diff
@@ -6,12 +6,21 @@ module ActiveSupport
   module Concurrency
     # A monitor that will permit dependency loading while blocked waiting for
     # the lock.
-    class LoadInterlockAwareMonitor < Monitor
+    class LoadInterlockAwareMonitor < Monitor # :nodoc:
       # Enters an exclusive section, but allows dependency loading while blocked
       def mon_enter
         mon_try_enter ||
           ActiveSupport::Dependencies.interlock.permit_concurrent_loads { super }
       end
+
+      def synchronize
+        Thread.handle_interrupt(Exception => :never) { mon_enter }
```

Comment:
How about extract `Exception => :never` as constant?
Now each `synchronize` call allocate new `Exception => :never` hash twice.

Reply:
Thanks for your review. I fixed.